misc/bstr.c

```diff
@@ -63,18 +63,25 @@ int bstrcasecmp(struct bstr str1, struct bstr str2)
 
 int bstrchr(struct bstr str, int c)
 {
-    for (int i = 0; i < str.len; i++)
-        if (str.start[i] == c)
-            return i;
-    return -1;
+    if (!str.len)
```

Comment:
`memchr` can handle `n == 0` and returns NULL in that case. IMO no need to check it beforehand.

Reply:
But it cannot handle `str.start == NULL`, so we check beforehand.

Reply:
You're right.